components/server/src/authorization/spicedb.ts

```diff
@@ -40,63 +34,48 @@ export function spiceDBConfigFromEnv(): SpiceDBClientConfig | undefined {
-        undefined,
-        clientOptions,
-    ) as Client;
-}
-
-export class CachingSpiceDBClientProvider implements SpiceDBClientProvider {
+export class SpiceDBClientProvider {
     private client: Client | undefined;
 
-    private readonly interceptors: grpc.Interceptor[] = [];
-
     constructor(
-        private readonly _clientConfig: SpiceDBClientConfig,
-        private readonly clientCallMetrics?: IClientCallMetrics | undefined,
-    ) {
-        if (this.clientCallMetrics) {
-            this.interceptors.push(createClientCallMetricsInterceptor(this.clientCallMetrics));
-        }
-    }
+        private readonly clientConfig: SpiceDBClientConfig,
+        private readonly interceptors: grpc.Interceptor[] = [],
+    ) {}
 
     getClient(): SpiceDBClient {
-        let client = this.client;
-        if (!client) {
-            client = spicedbClientFromConfig(this.clientConfig);
-        } else if (client.getChannel().getConnectivityState(true) !== grpc.connectivityState.READY) {
-            // (gpl): We need this check and explicit re-connect because we observe a ~120s connection timeout without it.
-            // It's not entirely clear where that timeout comes from, but likely from the underlying transport, that is not exposed by grpc/grpc-js
-            client.close();
-
-            log.warn("[spicedb] Lost connection to SpiceDB - reconnecting...");
-
-            client = spicedbClientFromConfig(this.clientConfig);
-        }
-        this.client = client;
-
-        return client.promises;
-    }
-
-    protected get clientConfig() {
-        const config = this._clientConfig;
-        if (this.interceptors) {
-            return {
-                ...config,
-                options: {
-                    interceptors: [...this.interceptors],
+        if (!this.client) {
+            this.client = v1.NewClient(
+                this.clientConfig.token,
+                this.clientConfig.address,
+                v1.ClientSecurity.INSECURE_PLAINTEXT_CREDENTIALS,
+                undefined, //
+                {
+                    // we ping frequently to check if the connection is still alive
+                    "grpc.keepalive_time_ms": 1000,
+                    "grpc.keepalive_timeout_ms": 1000,
+
+                    "grpc.max_reconnect_backoff_ms": 5000,
+                    "grpc.initial_reconnect_backoff_ms": 500,
+                    "grpc.service_config": JSON.stringify({
+                        methodConfig: [
+                            {
+                                name: [{}],
+                                retryPolicy: {
+                                    maxAttempts: 10,
+                                    initialBackoff: "0.1s",
+                                    maxBackoff: "5s",
+                                    backoffMultiplier: 2.0,
+                                    retryableStatusCodes: ["UNAVAILABLE", "DEADLINE_EXCEEDED"],
```

Comment:
From my testing it seems that keeping DEADLINE_EXCEEDED as retryable did not really help here: If we set timeouts to be lower than the max-retry-time, it just breaks the retries.
IMO we should retry for a max. time of e.g. 20s, and set the static timeouts to slightly higher, e.g. 22s. They are more a measure of making sure we clean up requests eventually in this context.

FYI nice-grpc also had the option to pass an abortSignal as deadline, which would be way more useful if we propagated it from the API down to here.

Reply:
From my tests it seems as if shorted deadline with this would be helpful when grpc-js needs to reconnect, as it would retry and in that use a new connection.

Reply:
So, I'd rather make the deadlin short (4 sec) and leave this here. But maybe we can roll this out and see how it behaves in prod, before going back and forth on the settings another time?

Reply:
I did not notice any difference: UNAVAILABLE is reported after ~4s, and my deadline was always higher than that. But ok with this it works. :shrug: 

In general it would be great to have more insights into this. But we should with the clientCallMetrics now, at least on the error side. :+1:

Reply:
> But maybe we can roll this out and see how it behaves in prod, before going back and forth on the settings another time?

It would be great to get more feedback in prod, but again: visibility.

In general I thought we had agreement on the config two days ago with respect to deadline and retry handling, that was why I was suprised by this because it differs from what I recall from that discussion. :slightly_smiling_face: 

But anyway, let's move on with this.

Reply:
FWIW I don't think I changed those bits since our conversation. It's a rabbit hole ...

Reply:
> I don't think I changed those bits since our conversation

:+1: but I thought we wanted to.

> I don't think I changed those bits since our conversation

Oh boy... :rabbit2: :laughing: